### PR TITLE
error handling using diagnostics and breaking at the correct point

### PR DESCRIPTION
**Pull request by sashu-shankar**
diagnostics/errors updated for non-YAML file input, missing required fields, unknown node types (pulumi issue), and CRD detection

CRD: 
Screen Shot 2020-09-11 at 12 25 44 PM

Missing Required Fields: 
Screen Shot 2020-09-11 at 12 24 56 PM

Lambda Updates to reflect these changes: https://github.com/pulumi/home/pull/1077

### CODE REVIEW

File: pkg/yaml2pcl/yaml2pcl.go [hunk 1]
```diff
@@ -25,11 +26,14 @@ import (
...
-		return "", err
+	if strings.Contains(filename, ".yaml") || strings.Contains(filename, ".yml") {
+		testFiles, err := parser.ParseFile(filename, parser.ParseComments)
+		if err != nil {
+			return "", err
+		}
+		return convert(*testFiles)
 	}
-	return convert(*testFiles)
+	return "", fmt.Errorf("input was not a YAML file: %s", filename)
```
Comment (lblackstone) [inline pkg/yaml2pcl/yaml2pcl.go:36]: This message is a little misleading. The logic requires that the file has a yaml extension, but doesn't verify the contents.

We should either update the message to indicate that the file didn't have the expected extension, or change the logic to try each file and bail out if it doesn't parse correctly.

File: pkg/yaml2pcl/yaml2pcl.go [hunk 2]
```diff
@@ -440,7 +469,8 @@ func walkToPCL(v Visitor, node ast.Node, totalPCL io.Writer, suffix string) erro
 			return err
 		}
 	default:
-		return fmt.Errorf("unexpected node type: " + n.Type().String())
+		return fmt.Errorf(fmt.Sprintf("unexpected node type: %s\n Please file an issue with the YAML used in order"+
+			"for this issue to get fixed: https://github.com/pulumi/kube2pulumi/issues", n.Type().String()))
```
Comment (lblackstone) [inline pkg/yaml2pcl/yaml2pcl.go:473]:
```suggestion
		return fmt.Errorf(fmt.Sprintf("unexpected node type: %s\n Please file an issue with the YAML input so we can take a look:"+
			"https://github.com/pulumi/kube2pulumi/issues", n.Type().String()))
```

**Review by lblackstone — CHANGES_REQUESTED** [review]
Looking good overall!

File: pkg/yaml2pcl/yaml2pcl.go (hunk 1 above)
Comment (EvanBoyle) [reply pkg/yaml2pcl/yaml2pcl.go:36]: I agree. We should be concerned about whether or not the file contents parse and not be opinionated about the extension

File: pkg/yaml2pcl/yaml2pcl.go (hunk 2 above)
Comment (EvanBoyle) [reply pkg/yaml2pcl/yaml2pcl.go:473]: Might as well take the user all the way there as well: 

https://github.com/pulumi/kube2pulumi/issues/new

**Review by lblackstone — APPROVED** [review]

**Review by lblackstone — APPROVED** [review]

**Pull request merged**